src/tty.c

```diff
@@ -56,84 +60,97 @@ void tty_open() {
       error("TTY: tty_open");
     }
   }
-} /* tty_open end */
-
-void tty_close() {
-  int stat;
-
-#ifdef TRACE
-  printf("TRACE: tty_close()\n");
-#endif
+}
 
+void tty_close(void)
+{
   if (TTY_Fd >= 0) {
     FD_CLR(TTY_Fd, &LispReadFds);
-    stat = close(TTY_Fd);
+    close(TTY_Fd);
 #ifdef TTYINT
     int_io_close(TTY_Fd);
 #endif
     TTY_Fd = (-1);
   }
-} /* tty_close end */
+}
 
-void TTY_get() {
+void tty_get(void)
+{
   char indata[256];
-  int count;
-
-#ifdef TRACE
-  printf("TRACE: tty_get()\n");
-#endif
 
   if ((TTY_Fd >= 0) && !DLTTYIn->state) {
     DLTTYIn->in_data = '\0'; /* Clear Previous Data */
 
-    if ((count = read(TTY_Fd, indata, 1)) == 1) {
+    if (read(TTY_Fd, indata, 1) == 1) {
       DLTTYIn->in_data = indata[0];
       DLTTYIn->state = 1;
     } else {
       error("TTY: tty_get");
     }
   }
-} /* TTY_get end */
+}
 
-void tty_put() {
-  int count;
+void tty_put(void)
+{
   char c;
 
-#ifdef TRACE
-  printf("TRACE: tty_put()\n");
-#endif
-
   if (TTY_Fd >= 0) {
     c = DLTTYPortCmd->outdata;
-    if ((count = write(TTY_Fd, &c, 1)) != 1) { error("TTY: tty_put()"); }
+    if (write(TTY_Fd, &c, 1) != 1) { error("TTY: tty_put()"); }
   }
-} /* tty_put end */
-
-void tty_breakon() {
-  int stat;
+}
 
-#ifdef TRACE
-  printf("TRACE: tty_breakon()\n");
-#endif
+speed_t tty_baudtosymbol(short aBaud)
+{
+  if (aBaud == 0) return (B50);
+  if (aBaud == 1) return (B75);
+  if (aBaud == 2) return (B110);
+  if (aBaud == 3) return (B134);
+  if (aBaud == 4) return (B150);
+  if (aBaud == 5) return (B300);
+  if (aBaud == 6) return (B600);
+  if (aBaud == 7) return (B1200);
+  if (aBaud == 10) return (B2400);
+  if (aBaud == 12) return (B4800);
+  if (aBaud == 14) return (B9600);
+  if (aBaud == 15) return (EXTA);
+  return (-1);
+}
 
-  if (TTY_Fd >= 0) { stat = ioctl(TTY_Fd, TIOCSBRK, 0); }
-} /* tty_breakon end */
+void tty_setbaudrate(void)
+{
+  speed_t baudrate;
 
-void tty_breakoff() {
-  int stat;
+  if (TTY_Fd >= 0) {
+    if ((baudrate = tty_baudtosymbol(DLTTYOut->line_speed)) != -1) {
+        struct termios options;
+        tcgetattr(TTY_Fd, &options);
+        cfsetispeed(&options, baudrate);
```

Comment:
Nit: `cfsetspeed(&options, baudrate)` does both input and output speed at once, since we always make them the same.

Reply:
It ends up that `cfsetspeed` is a BSD 4.4 extension and not supported by Solaris. It also ends up that Solaris apparently doesn't have `cfmakeraw` either, so I've expanded that and force pushed.